tests/console/zypper_lr_validate.pm

```diff
@@ -15,6 +15,326 @@ use strict;
 use testapi;
 use utils;
 
+sub validatelr {
+    my ($args) = @_;
+
+    my $alias           = $args->{alias} || "";
+    my $product         = $args->{product};
+    my $product_channel = $args->{product_channel} || "";
+    my $version         = $args->{version};
+    if (get_var('ZDUP')) {
+        $version = "";
+    }
+    if (get_var('FLAVOR') =~ m{SAP}) {
+        $version .= "-SAP";
+    }
+    # Live patching and other modules are not per-service pack channel model,
+    # so use major version to validate their repos
+    if ($product eq 'SLE-Live') {
+        $product = 'SLE-Live-Patching';
+        $version = '12';
+    }
+    if ($product eq 'SLE-ASMM') {
+        $product = 'SLE-Module-Adv-Systems-Management';
+        $version = '12';
+    }
+    if ($product eq 'SLE-CONTM') {
+        $product = 'SLE-Module-Containers';
+        $version = '12';
+    }
+    if ($product eq 'SLE-HPCM') {
+        $product = 'SLE-Module-HPC';
+        $version = '12';
+    }
+    if ($product eq 'SLE-LGM') {
+        $product = 'SLE-Module-Legacy';
+        $version = '12';
+    }
+    if ($product eq 'SLE-PCM') {
+        $product = 'SLE-Module-Public-Cloud';
+        $version = '12';
+    }
+    if ($product eq 'SLE-TCM') {
+        $product = 'SLE-Module-Toolchain';
+        $version = '12';
+    }
+    if ($product eq 'SLE-WSM') {
+        $product = 'SLE-Module-Web-Scripting';
+        $version = '12';
+    }
+    if ($product eq 'SLE-PHUB') {
+        $product = 'SUSE-PackageHub-';
+    }
+    # LTSS version is included in its product name
+    # leave it as empty to match the regex
+    if ($product =~ /LTSS/) {
+        $version = '';
+    }
+    diag "validatelr alias:$alias product:$product cha:$product_channel version:$version";
+
+    # Repo is checked for enabled/disabled state. If the information about the
+    # expected state is not delivered to validatelr(), we use some heuristics to
+    # determine the expected state: If the installation medium is a physical
+    # medium and the system is registered to SCC the repo should be disabled
+    # if the system is SLE 12 SP2 and later; enabled otherwise, see PR#11460 and
+    # FATE#320494.
+    my $scc_install_sle12sp2 = check_var('SCC_REGISTER', 'installation') and sle_version_at_least('12-SP2');
+    my $enabled_repo;
+    if ($args->{enabled_repo}) {
+        $enabled_repo = $args->{enabled_repo};
+    }
+    # bsc#1012258, bsc#793709: USB repo is disabled as the USB stick will be
+    # very likely removed from the system.
+    elsif ($args->{uri} =~ m{(cd|dvd|hd):///.*usb-}) {
+        $enabled_repo = 'No';
```

Comment:
I understand that you just copy/paste this .... but still it's nightmare. Text representation of boolean is beyond my understanding